lib/Frontend/Frontend.cpp

```diff
@@ -818,6 +818,26 @@ SourceFile *CompilerInstance::getIDEInspectionFile() const {
   return evaluateOrDefault(eval, IDEInspectionFileRequest{mod}, nullptr);
 }
 
+static inline bool isPCHFilenameExtension(StringRef path) {
+  return llvm::sys::path::extension(path)
+    .endswith(file_types::getExtension(file_types::TY_PCH));
```

Comment:
Can we do this based on how clang will actually handle the file instead of using the file extension here?  My concern is that `.pch` is a valid header filename and it's not rare to see in practice.

Reply:
The swift makes lots of assumptions using file extensions regarding the file passed through import-objc-header so making this more resilient doesn't change anything.

On one side, we can improve all pch handling but we also kind of want to make all of them modules. Hence I think just check file extension is the safest thing to do for release branch.